morpheus/_lib/doca/src/doca_source.cpp

```diff
@@ -68,158 +68,290 @@ std::optional<uint32_t> ip_to_int(std::string const& ip_address)
+                    cudf::column_view(cudf::device_span<const int32_t>(ether_type_out_d[idxs])),
+                    cudf::column_view(cudf::device_span<const int32_t>(next_proto_id_out_d[idxs])),
+                    cudf::column_view(cudf::device_span<const uint32_t>(timestamp_out_d[idxs])),
+                });
+        }
+
+        auto exit_condition = std::make_unique<morpheus::doca::DocaMem<uint32_t>>(m_context, 1, DOCA_GPU_MEM_TYPE_GPU_CPU);
         DOCA_GPUNETIO_VOLATILE(*(exit_condition->cpu_ptr())) = 0;
 
         auto cancel_thread = std::thread([&] {
             while (output.is_subscribed()) {}
             DOCA_GPUNETIO_VOLATILE(*(exit_condition->cpu_ptr())) = 1;
         });
 
-        while (output.is_subscribed())
-        {
-            if (DOCA_GPUNETIO_VOLATILE(*(exit_condition->cpu_ptr())) == 1)
-            {
+        while (output.is_subscribed()) {
+
+            if (DOCA_GPUNETIO_VOLATILE(*(exit_condition->cpu_ptr())) == 1) {
                 output.unsubscribe();
                 continue;
             }
 
-            morpheus::doca::packet_receive_kernel(m_rxq->rxq_info_gpu(),
-                                                  m_semaphore->gpu_ptr(),
-                                                  m_semaphore->size(),
-                                                  semaphore_idx_d.data(),
-                                                  packet_count_d.data(),
-                                                  payload_buffer_d.data(),
-                                                  payload_size_total_d.data(),
-                                                  payload_sizes_d.data(),
-                                                  src_mac_out_d.data(),
-                                                  dst_mac_out_d.data(),
-                                                  src_ip_out_d.data(),
-                                                  dst_ip_out_d.data(),
-                                                  src_port_out_d.data(),
-                                                  dst_port_out_d.data(),
-                                                  tcp_flags_out_d.data(),
-                                                  ether_type_out_d.data(),
-                                                  next_proto_id_out_d.data(),
-                                                  timestamp_out_d.data(),
-                                                  exit_condition->gpu_ptr(),
-                                                  rmm::cuda_stream_default);
-
-            cudaStreamSynchronize(rmm::cuda_stream_default);
-
-            auto packet_count = packet_count_d.value(rmm::cuda_stream_default);
-
-            if (packet_count == 0)
-            {
-                continue;
+            /*
+             * Can't run the received kernel in persistent mode for a weirdness in the rxcpp/mrc framework.
```

Comment:
Please open an issue in the Morpheus github repository with this comment and any other pertinent information, and remove this comment from the code.

Reply:
https://github.com/nv-morpheus/Morpheus/issues/1557